test/cql-pytest/cassandra_tests/validation/entities/json_test.py

```diff
@@ -929,6 +924,15 @@ def testToJsonFct(cql, test_keyspace):
             execute(cql, table, "INSERT INTO %s (k, durationval) VALUES (?, P1Y1M2DT10H5M)", 0)
             assert_rows(execute(cql, table, "SELECT k, toJson(durationval) FROM %s WHERE k = ?", 0), [0, "\"1y1mo2d10h5m\""])
 
+# Reproduces issue #7997
+def testTimestampToJson(cql, test_keyspace):
+    with create_type(cql, test_keyspace, "(a int, b uuid, c set<text>)") as type_name:
+        with create_table(cql, test_keyspace, "(" +
+                "k int PRIMARY KEY, " +
+                "timestampval timestamp)") as table:
+            execute(cql, table, "INSERT INTO %s (k, timestampval) VALUES (?, ?)", 0, datetime(2014, 1, 1, 0, 0, 0))
+            assert_rows(execute(cql, table, "SELECT k, toJson(timestampval) FROM %s WHERE k = ?", 0), [0, "\"2014-01-01T00:00:00.000Z\""])
+
```

Comment:
Thanks. I run cassandra in docker container, is it possible to run this script using cassandra in container?

Reply:
The run-cassandra script is not much more than a convenience script that runs Cassandra for you in a temporary directory, and then runs "pytest" to run all the tests. If you know how to run Cassandra yourself (via docker or however you're used to running Cassandra), then just run it yourself, and then do something like
```
cd cqlpytest
pytest json_test.py::some_function_name
```

If your Cassandra isn't listening on localhost:9042 but rather on some other IP address or port (possibly even on another machine), just pass the "--host" and/or "--port" option to pytest, e.g.,
```
pytest --host 1.2.3.4 --port 4567 json_test.py::some_function_name
```

It's very easy and convenient.

Reply:
thanks, will try

Reply:
I've got an idea how to fix problem with JSON string format but not sure we should do it in this patch...

Reply:
I've got an idea how to fix JSON format, will add to this patch if you don't want to split these fixes into different patches

Reply:
I've added a function to support JSON timestamp string format. It's the simplest way I found.
I tried to add t`o_json_string() `method to `timestamp_date_base_class` and maybe it would look better a little but it required a lot of linked changes and I've chosen another way.
All tests are passed locally, including our json tests on Cassandra as we discussed above